dipy/workflows/reconst.py

```diff
@@ -21,6 +21,330 @@
+        save_metrics :
+            List of metrics to save.
+            Possible values: rtop, laplacian_signal, msd, qiv, rtap, rtpp
+            (default: [] (all))
+        out_dir : string, optional
+            Output directory (default: input file directory)
+        out_rtop : string, optional
+            Name of the rtop to be saved
+        out_lapnorm : string, optional
+            Name of the norm of laplacian signal to be saved
+        out_msd : string, optional
+            Name of the msd to be saved
+        out_qiv : string, optional
+            Name of the qiv to be saved
+        out_rtap : string, optional
+            Name of the rtap to be saved
+        out_rtpp : string, optional
+            Name of the rtpp to be saved
+        """
+        io_it = self.get_io_iterator()
+        for dwi, bval, bvec, out_rtop, out_lapnorm, out_msd, out_qiv, out_rtap, out_rtpp in io_it:
+
+            logging.info('Computing DTI metrics for {0}'.format(dwi))
+            img = nib.load(dwi)
+            data = img.get_data()
+            affine = img.affine
+            bvals, bvecs = read_bvals_bvecs(bval, bvec)
+
+            gtab = gradient_table(bvals=bvals, bvecs=bvecs,
+                                  small_delta=small_delta,
+                                  big_delta=big_delta, b0_threshold=50)
+
+            if model_type is None:
+                model_type = 'both'
+
+            if not save_metrics:
+                save_metrics = ['rtop', 'laplacian_signal', 'msd', 'qiv', 'rtap', 'rtpp']
+
+            # print('data.shape (%d, %d, %d, %d)' % data.shape)
+
+            radial_order = 6
+            
+            # How do I save the rtop in a useful form? A picture? A numpy array?
+            # Save them as nifti image
+
+            if model_type is 'laplacian':
+                map_model_aniso = mapmri.MapmriModel(gtab, radial_order=radial_order,
+                                                     laplacian_regularization=True,
+                                                     laplacian_weighting=.2)
+                mapfit_aniso = map_model_aniso.fit(data)
+
+            elif model_type is 'positivity':
+
+                map_model_aniso = mapmri.MapmriModel(gtab, radial_order=radial_order,
+                                                     laplacian_regularization=False,
+                                                     positivity_constraint=True)
+                mapfit_aniso = map_model_aniso.fit(data)
+
+            elif model_type is 'both':
+                map_model_aniso = mapmri.MapmriModel(gtab, radial_order=radial_order,
+                                                     laplacian_regularization=True,
+                                                     laplacian_weighting=.05,
+                                                     positivity_constraint=True)
+                mapfit_aniso = map_model_aniso.fit(data)
+
+            # Not sure where to get affine or metadata?
```

Comment:
Is this still an issue? Getting the affine? You can get it from the `data_file`.

As an aside: if you have questions, it's better to ask them in the PR interface than put them as comments in your code!

Reply:
Ah thanks for pointing that out. I added those notes in the code to ask you at our weekly sessions and forgot to delete them.